### PR TITLE
[MM-46603] P1 - Improvements in handling '0 posts' channels

### PR DESCRIPTION
**Pull request by plant99**
#### Summary

1) Currently, channels with `0 posts` in a given time period are ignored by inactive channels
2) `channel_join` event is considered as activity to filter out new channels.

This PR changes order of SQL joins to 
1) Include channels with 0 posts
2) Filter out new channels with `Channel.CreateAt`

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-46603

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: store/sqlstore/channel_store.go [hunk 1]
```diff
@@ -4337,48 +4337,45 @@ func (s SqlChannelStore) GetTopInactiveChannelsForTeamSince(teamID string, userI
...
-				count(Posts.Id) AS MessageCount,
-				max(Posts.CreateAt) AS LastActivityAt
+				COALESCE(count(Posts.Id), 0) AS MessageCount,
+				COALESCE(max(Posts.CreateAt), 0) AS LastActivityAt
 			FROM
-				Posts
-				LEFT JOIN PublicChannels on Posts.ChannelId = PublicChannels.Id
+				PublicChannels
+				LEFT JOIN Posts on Posts.ChannelId = PublicChannels.Id AND Posts.Type = '' AND Posts.CreateAt > ? AND Posts.DeleteAt = 0
+				LEFT JOIN Channels on Channels.Id = PublicChannels.Id
```
Comment (mkraft) [inline store/sqlstore/channel_store.go:4349]: What's the reason for joining `Channels` into this subquery that already has `PublicChannels`?

Comment (plant99) [reply store/sqlstore/channel_store.go:4349]: Channels.CreateAt is used to check if the channel was created before a particular time. This field is absent in `PublicChannels`

**Review by BenCookie95 — APPROVED** [review]
LGTM, did we performance test this?

**Comment by amyblais** [comment]
@plant99 The CI failed here.

**Review by mkraft — APPROVED** [review]

**Review by furqanmlk — APPROVED** [review]
Working fine

**Pull request merged**